### ISSUE TITLE
Kubernetes operator won't register: creating operator authkey

### ISSUE THREAD

**Issue by dmangot** [tailscale/tailscale#12059]
### What is the issue?

Brand new K3s cluster on a brand new Tailscale account.  I have a phone and laptop connected to the Tailnet already.

I followed the instructions at https://tailscale.com/kb/1236/kubernetes-operator#installation

When trying to apply the manifest, I get:

```
$ kubectl logs operator-84d65c9977-zsk8f --namespace tailscale
{"level":"fatal","ts":"2024-05-08T04:59:55Z","logger":"startup","msg":"creating operator authkey: Post \"https://api.tailscale.com/api/v2/tailnet/-/keys\": Post \"https://login.tailscale.com/api/v2/oauth/token\": dial tcp: lookup login.tailscale.com on 10.43.0.10:53: server misbehaving","stacktrace":"main.initTSNet\n\ttailscale.com/cmd/k8s-operator/operator.go:191\nmain.main\n\ttailscale.com/cmd/k8s-operator/operator.go:94\nruntime.main\n\truntime/proc.go:271"}
```

operator.go:94
```
	s, tsClient := initTSNet(zlog)
```

operator.go:191
```
if err != nil {
				startlog.Fatalf("creating operator authkey: %v", err)
			}
```

### Steps to reproduce

Creating the policy seems to have worked, because when I was creating the Oauth credentials, I was able to set the tag correctly.  Using the "manifest" method instead of the Helm chart so I could learn more.  

I have also quoted the entries in the manifest file so they are strings.

```
$ k3s --version
k3s version v1.29.4+k3s1 (94e29e2e)
go version go1.21.9
```

### Are there any recent changes that introduced the issue?

This is the 1st time trying on a brand new account and K3s install.

### OS

Linux

### OS version

Rocky Linux release 9.3

### Other software

k3s version v1.29.4+k3s1 (94e29e2e)
go version go1.21.9

Don't have tailscale installed locally to run `bugreport`

**Comment by irbekrm** [comment]
> lookup login.tailscale.com on 10.43.0.10:53 server misbehaving

That error suggests that the DNS lookup failed.
Is 10.43.0.10:53 your cluster DNS `Service`? Does that seem to be running allright? What happens if you `dig @10.43.0.10 tailscale.com` from within the cluster?

**Comment by dmangot** [comment]
Yes, too many changes happening here with the new setup.  Sorry for the trouble.

**Comment by irbekrm** [comment]
No worries, hopefully it's working now!

**Comment by Mavo81** [comment]
Hi, @dmangot ,
how did you resolve this problem? I have the exact same error message.

**Comment by dmangot** [comment]
@Mavo81 this was part of revamping my homelab.  One of the things I did was move DNS into K8s (Technitium).  However, on the K8s server (in `resolv.conf`) it was still pointing to the server I decommissioned.  Since the K3s CoreDNS points at the resolver on the host, it was basically pointed at something that went nowhere.  When I fixed `resolv.conf` and restarted CoreDNS in the main namespace, everything started working properly.
